nutanix/resource_nutanix_karbon_cluster_worker_pool.go

```diff
@@ -0,0 +1,400 @@
+	}
+	if err = d.Set("node_os_version", nodepool.NodeOSVersion); err != nil {
+		return diag.Errorf("error setting node_os_version for nke Worker Node Pool %s: %s", d.Id(), err)
+	}
+	if err = d.Set("num_instances", nodepool.NumInstances); err != nil {
+		return diag.Errorf("error setting num_instances for nke Worker Node Pool %s: %s", d.Id(), err)
+	}
+	if err = d.Set("nodes", nodes); err != nil {
+		return diag.Errorf("error setting nodes for nke Worker Node Pool %s: %s", d.Id(), err)
+	}
+	if err = d.Set("labels", nodepool.Labels); err != nil {
+		return diag.Errorf("error setting labels for nke Worker Node Pool %s: %s", d.Id(), err)
+	}
+	if err = d.Set("ahv_config", flattenAHVConfig(nodepool.AHVConfig)); err != nil {
+		return diag.Errorf("error setting ahv_config for nke Worker Node Pool %s: %s", d.Id(), err)
+	}
+	return nil
+}
+func resourceNutanixKarbonWorkerNodePoolUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*Client)
+	conn := client.KarbonAPI
+
+	karbonClsName := d.Get("nke_cluster_name")
+	workerName := d.Get("name")
+	resp, err := conn.Cluster.GetKarbonCluster(karbonClsName.(string))
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+	nodepool, err := conn.Cluster.GetKarbonClusterNodePool(*resp.Name, workerName.(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if d.HasChange("num_instances") {
+		old, new := d.GetChange("num_instances")
+
+		if old.(int) > new.(int) {
+			amountOfNodes := old.(int) - new.(int)
+			scaleDownRequest := &karbon.ClusterScaleDownIntentInput{
+				Count: int64(amountOfNodes),
+			}
+			karbonClusterActionResponse, err := client.KarbonAPI.Cluster.ScaleDownKarbonCluster(
+				*resp.Name,
+				*nodepool.Name,
+				scaleDownRequest,
+			)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			err = WaitForKarbonCluster(ctx, client, 0, karbonClusterActionResponse.TaskUUID, d.Timeout(schema.TimeoutUpdate))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		} else {
+			amountOfNodes := new.(int) - old.(int)
+			scaleUpRequest := &karbon.ClusterScaleUpIntentInput{
+				Count: int64(amountOfNodes),
+			}
+			karbonClusterActionResponse, err := client.KarbonAPI.Cluster.ScaleUpKarbonCluster(
+				*resp.Name,
+				*nodepool.Name,
+				scaleUpRequest,
+			)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			err = WaitForKarbonCluster(ctx, client, 0, karbonClusterActionResponse.TaskUUID, d.Timeout(schema.TimeoutUpdate))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+	if d.HasChange("labels") {
+		old, new := d.GetChange("labels")
+		updateLabelRequest := &karbon.UpdateWorkerNodeLabels{}
+
+		newMap := new.(map[string]interface{})
+		oldMap := old.(map[string]interface{})
+		addLabelMap := map[string]string{}
+		removeLabel := []string{}
+
+		// check any new is label is added.
+		for key := range newMap {
+			if _, ok := oldMap[key]; ok {
+				continue
```

Comment:
for old key, value can be changed right ?

Reply:
No, we can't add same key and can't update the value. That's the karbon UI behaviour .